src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/Serialization/SymbolSpecification.cs

```diff
@@ -179,7 +180,7 @@ private void PopulateModifierListFromXElement(XElement modifierListElement)
             }
         }
 
-        public class SymbolKindOrTypeKind
+        public class SymbolKindOrTypeKind : IEquatable<SymbolKindOrTypeKind>
```

Comment:
Not from you, but is there a reason why this, `AccessibilityKind`, and `ModifierKind` aren't structs?